resteasy-core/src/main/java/org/jboss/resteasy/plugins/providers/MultiValuedArrayParamConverter.java

```diff
@@ -0,0 +1,189 @@
+package org.jboss.resteasy.plugins.providers;
+
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Array;
+import java.lang.reflect.Method;
+
+import javax.annotation.Priority;
+import javax.ws.rs.ext.ParamConverter;
+import javax.ws.rs.ext.Provider;
+
+import org.jboss.resteasy.core.StringParameterInjector;
+import org.jboss.resteasy.resteasy_jaxrs.i18n.Messages;
+
+/**
+ * @author Marek Kopecky mkopecky@redhat.com
+ * @author Ron Sigal rsigal@redhat.com
+ */
+@Provider
+@Priority(Integer.MAX_VALUE)
+public class MultiValuedArrayParamConverter implements ParamConverter<Object>
+{
+   private Class<?> rawType;
+   private StringParameterInjector stringParameterInjector;
+   private String separator;
+
+   public MultiValuedArrayParamConverter(Class<?> rawType, Annotation[] annotations, StringParameterInjector stringParameterInjector, String separator)
+   {
+      this.rawType = rawType;
+      this.stringParameterInjector = stringParameterInjector;
+      this.separator = separator;
+   }
+
+   @Override
+   public String toString(Object value)
+   {
+      if (value == null)
+      {
+         return null;
+      }
+      if (!value.getClass().isArray())
+      {
+         throw new RuntimeException(Messages.MESSAGES.expectedArray(value.getClass().getName()));
+      }
+      Object[] values = new Object[Array.getLength(value)];
+      if (values.length == 0)
+      {
+         return "";
+      }
+      if (rawType.getComponentType().isPrimitive())
+      {
+         copyPrimitiveArray(values, value);
```

Comment:
If I understand well, `copyPrimitiveArray(...)` method is to filter array elements whose type is not equal to the type that this param converter can handle (`rawType.getComponentType()`) right ?

But if I'm right in which situation such case can happen ?

For example if I have a proxy interface like that:
```
public interface Client{
	Response getItems(int[]itemsId);
}
```
When user will use proxy instance to invoke `getItems(int[]itemsId)` he will only be able to provide a int[] parameter not an Object nor Object[]. So filtering done by the param converter will be useless. 

And by the way if we are doing this kind of filtering for primitive type we have to do it for Object type as well.

As I said above I don't know in which situation this case can happen but if we want to add a check for it, the same way we are checking for parameter validity by verifying that it is an array as expected, will it not be good to also test for the component type ?
```

else if (rawType.getComponentType().isAssignableFrom(value.getClass().getComponentType()))
{
 throw new RuntimeException()
}
```

What do you think? Am I missing something?

Reply:
Well, the point of copyPrimitiveArray() was to work around Java's unfortunate distinction between primitive values and Objects. So toString() ends up with an array of Objects in both cases. I guess the alternative would be to have a different stringify() for each primitive type.

Reply:
When reading your answer first time I did not understand your point very well.
Now it's clearer and it also makes me understand why you make this class extends from `ParamConverter<Object>` instead of `ParamConverter<Object[]>`.
I forgot this java distiction about primitive  array and object array.
Thanks :+1:

Reply:
Great!!